tpm2/tpm2.go

```diff
@@ -0,0 +1,1247 @@
+func EvictControl(rw io.ReadWriter, owner, tmpHandle, persistantHandle Handle) error {
+	cmd, err := encodeEvictControl(owner, tmpHandle, persistantHandle)
+	if err != nil {
+		return err
+	}
+	_, err = runCommand(rw, tagSessions, cmdEvictControl, tpmutil.RawBytes(cmd))
+	return err
+}
+
+// ContextSave returns an encrypted version of the session, object or sequence
+// context for storage outside of the TPM.
+func ContextSave(rw io.ReadWriter, handle Handle) ([]byte, error) {
+	return runCommand(rw, tagNoSessions, cmdContextSave, handle)
+}
+
+func decodeLoadContext(in []byte) (Handle, error) {
+	var handle Handle
+	err := tpmutil.Unpack(in, &handle)
+	if err != nil {
+		return 0, err
+	}
+	return Handle(handle), nil
+}
+
+// ContextLoad reloads context data created by ContextSave.
+func ContextLoad(rw io.ReadWriter, saveArea []byte) (Handle, error) {
+	resp, err := runCommand(rw, tagNoSessions, cmdContextLoad, tpmutil.RawBytes(saveArea))
+	if err != nil {
+		return 0, err
+	}
+	handle, err := decodeLoadContext(resp)
+	if err != nil {
+		return 0, fmt.Errorf("decoding LoadContext response: %v", err)
+	}
+	return handle, nil
+}
+
+func encodeIncrementNv(handle Handle, authString string) ([]byte, error) {
+	auth, err := encodePasswordAuthArea(authString, HandlePasswordSession)
+	if err != nil {
+		return nil, err
+	}
+	out, err := tpmutil.Pack(handle, handle, []byte(nil))
+	if err != nil {
+		return nil, err
+	}
+	out = append(out, auth...)
+	return out, nil
+}
+
+// NVIncrement increments a counter NV index.
+func NVIncrement(rw io.ReadWriter, handle Handle, authString string) error {
+	cmd, err := encodeIncrementNv(handle, authString)
+	if err != nil {
+		return err
+	}
+	_, err = runCommand(rw, tagSessions, cmdIncrementNvCounter, tpmutil.RawBytes(cmd))
+	return err
+}
+
+func encodeUndefineSpace(owner, handle Handle) ([]byte, error) {
+	auth, err := encodePasswordAuthArea("", HandlePasswordSession)
+	if err != nil {
+		return nil, err
+	}
+	out, err := tpmutil.Pack(owner, handle, uint16(0))
+	if err != nil {
+		return nil, err
+	}
+	out = append(out, auth...)
+	return out, nil
+}
+
+// NVUndefineSpace removes an index from TPM's NV storage.
+func NVUndefineSpace(rw io.ReadWriter, owner, handle Handle) error {
+	cmd, err := encodeUndefineSpace(owner, handle)
+	if err != nil {
+		return err
+	}
+	_, err = runCommand(rw, tagSessions, cmdUndefineSpace, tpmutil.RawBytes(cmd))
+	return err
+}
+
+func encodeDefineSpace(owner, handle Handle, authString string, attributes uint32, policy []byte, dataSize uint16) ([]byte, error) {
+	pw, err := encodePasswordData(authString)
+	if err != nil {
+		return nil, err
+	}
+	auth, err := encodePasswordAuthArea("", HandlePasswordSession)
+	if err != nil {
+		return nil, err
+	}
+	out1, err := tpmutil.Pack(owner, []byte(nil))
+	if err != nil {
+		return nil, err
+	}
+	hashAlg := AlgSHA1
+	sizeNvArea := uint16(2*int(unsafe.Sizeof(owner)) + 3*int(unsafe.Sizeof(dataSize)) + len(policy))
```

Comment:
`sizeNvArea` -> `sizeNVArea` (fully capitalize initialisms for Go style)

Also, please add a comment about why this computation is the way it is.

Reply:
Done for naming